src/app/shared/components/button/button.component.ts

```diff
@@ -1,14 +1,17 @@
 import { Component, Input } from '@angular/core';
-import {scheduleReadableStreamLike} from "rxjs/internal/scheduled/scheduleReadableStreamLike";
+import { scheduleReadableStreamLike } from 'rxjs/internal/scheduled/scheduleReadableStreamLike';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrl: './button.component.scss',
 })
 export class ButtonComponent {
+  //TODO: that does not sound as a good approach to me, as in the end there will be tens or hundreds of classes.
```

Comment:
please read my TODOs
this component's idea looks like something that will be extremely hard to maintain with the project growth